### PR TITLE
Add Documentation for lightly/loss subpackage.

### LINKED ISSUE #1673: Cleanup docstrings in lightly/loss subpackage.
See #1216 for information

### PR DESCRIPTION
**Pull request by ayush22iitbhu**
Fixes #1673 
I have done the following:

1. Added documentation wherever necessary. For example, many functions had the error description part missing in their docstrings.
2. Corrected any grammatical error that I found
3. Added more comments for clarity

### CODE REVIEW

File: lightly/loss/barlow_twins_loss.py [hunk 1]
```diff
@@ -30,37 +30,53 @@ class BarlowTwinsLoss(torch.nn.Module):
 
     def __init__(self, lambda_param: float = 5e-3, gather_distributed: bool = False):
         """Lambda param configuration with default value like in [0]
+            Initializes the BarlowTwinsLoss with the specified parameters.
```
Comment (guarin) [inline lightly/loss/barlow_twins_loss.py:33]: Please follow the google style guide for docstrings: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings:

```
    """Fetches rows from a Smalltable.

    Retrieves rows pertaining to the given keys from the Table instance
    represented by table_handle.  String keys will be UTF-8 encoded.

    Args:
        table_handle: An open smalltable.Table instance.
        keys: A sequence of strings representing the key of each table
          row to fetch.  String keys will be UTF-8 encoded.
        require_all_keys: If True only rows with values set for all keys will be
          returned.

    Returns:
        A dict mapping keys to the corresponding table row data
        fetched. Each row is represented as a tuple of strings. For
        example:

        {b'Serak': ('Rigel VII', 'Preparer'),
         b'Zim': ('Irk', 'Invader'),
         b'Lrrr': ('Omicron Persei 8', 'Emperor')}

        Returned keys are always bytes.  If a key from the keys argument is
        missing from the dictionary, then that row was not found in the
        table (and require_all_keys must have been False).

    Raises:
        IOError: An error occurred accessing the smalltable.
    """
```

Specifically, there should be an empty line between the first and second sentence in the docstring. This also applies for all the other docstrings.

File: lightly/loss/barlow_twins_loss.py [hunk 2]
```diff
@@ -77,8 +93,21 @@ def forward(self, z_a: torch.Tensor, z_b: torch.Tensor) -> torch.Tensor:
 def _normalize(
     z_a: torch.Tensor, z_b: torch.Tensor
 ) -> Tuple[torch.Tensor, torch.Tensor]:
-    """Helper function to normalize tensors along the batch dimension."""
+    """
+    Helper function to normalize tensors along the batch dimension.
```
Comment (guarin) [inline lightly/loss/barlow_twins_loss.py:97]: Just FYI, for private functions (starting with an underscore) it is not necessary to write very detailed docstrings. Exceptions are complex functions with a lot of logic.

File: lightly/loss/barlow_twins_loss.py [hunk 3]
```diff
@@ -30,37 +30,53 @@ class BarlowTwinsLoss(torch.nn.Module):
...
-                "gather_distributed is True but torch.distributed is not available. "
+                "gather_distributed is True but torch.distributed is not available."
                 "Please set gather_distributed=False or install a torch version with "
                 "distributed support."
             )
 
     def forward(self, z_a: torch.Tensor, z_b: torch.Tensor) -> torch.Tensor:
-        # normalize repr. along the batch dimension
+        """
+        Computes the Barlow Twins loss for the given projections.
```
Comment (guarin) [inline lightly/loss/barlow_twins_loss.py:59]:
```suggestion
        """Computes the Barlow Twins loss for the given projections.
```

Same for the other docstrings.

**Review by guarin — CHANGES_REQUESTED** [review]
Thanks a lot for this extensive PR! Left some comments regarding minor formatting issues but otherwise this looks very nice!

File: lightly/loss/barlow_twins_loss.py (hunk 1 above)
Comment (ayush22iitbhu) [reply lightly/loss/barlow_twins_loss.py]: @guarin done

File: lightly/loss/barlow_twins_loss.py (hunk 2 above)
Comment (ayush22iitbhu) [reply lightly/loss/barlow_twins_loss.py]: @guarin done.

File: lightly/loss/barlow_twins_loss.py (hunk 3 above)
Comment (ayush22iitbhu) [reply lightly/loss/barlow_twins_loss.py]: @guarin done.

**Comment by ayush22iitbhu** [comment]
@guarin it is ready for review.

**Review by guarin — APPROVED** [review]
This LGTM! Thanks for the contribution!

**Pull request merged**